File: app/src/main/java/com/example/pearvideoclient/channel/CategoryFragment.java
package com.example.pearvideoclient.channel;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.example.pearvideoclient.CommonCallBack;
import com.example.pearvideoclient.Constants;
import com.example.pearvideoclient.R;
import com.example.pearvideoclient.channel.adapter.ContListAdapter;
import com.example.pearvideoclient.entity.ContEntity;
import com.scwang.smartrefresh.layout.api.RefreshLayout;

import java.util.ArrayList;
import java.util.List;

/**
 * @Description: java类作用描述
 * @Author: jiazhu
 * @Date: 2019-08-29 15:51
 * @ClassName: CategoryFragment
 */
public class CategoryFragment extends Fragment {
    private RecyclerView mRvCategoryContsList;
    private RefreshLayout mRefreshLayout;
    private String mName;
    private String mCategoryId;
    private ContListAdapter mContListAdapter;

    private CommonCallBack<Integer, Void> refreshCallBack;

    public void setRefreshCallBack(CommonCallBack<Integer, Void> refreshCallBack) {
        this.refreshCallBack = refreshCallBack;
    }

    public static CategoryFragment newInstance(String name, String categoryId) {
        Bundle bundle = new Bundle();
        bundle.putString("name", name);
        bundle.putString("categoryId", categoryId);
        CategoryFragment fragment = new CategoryFragment();
        fragment.setArguments(bundle);
        return fragment;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_category_conts, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        initView(view);
        initData();
    }

    private void initView(View view) {
        mRefreshLayout = view.findViewById(R.id.refresh_layout);
        mRvCategoryContsList = view.findViewById(R.id.rv_category_cont_list);
    }

    private void initData() {
        if (getArguments() != null) {
            mName = getArguments().getString("name");
            mCategoryId = getArguments().getString("categoryId");
        }

        AppCompatActivity activity = (AppCompatActivity) getActivity();
        mRvCategoryContsList.setLayoutManager(new LinearLayoutManager(activity, RecyclerView.VERTICAL, false));
        mContListAdapter = new ContListAdapter(new ArrayList<>());
        mRvCategoryContsList.setAdapter(mContListAdapter);

        mRefreshLayout.setOnRefreshListener(refreshLayout -> {
            if (refreshCallBack != null) {
                refreshCallBack.todo(Constants.LOAD_REFRESH);
            }
        });
        mRefreshLayout.setOnLoadMoreListener(refreshLayout -> {
            if (refreshCallBack != null) {
                refreshCallBack.todo(Constants.LOAD_MORE);
            }
        });
    }

    public void replaceData(List<ContEntity> data) {
        mContListAdapter.replaceData(data);
    }

    public void addData(List<ContEntity> data) {
        mContListAdapter.addData(data);
    }

    public void loadMoreFinish(boolean isSuccess) {
        mRefreshLayout.finishLoadMore(isSuccess);
    }

    public void refreshFinish(boolean isSuccess) {
        mRefreshLayout.finishRefresh(isSuccess);
    }

    public boolean hasData() {
        return mContListAdapter.getData().size() > 0;
    }
}

File: app/src/main/java/com/example/pearvideoclient/channel/CategoryFragment.kt
package com.example.pearvideoclient.channel

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

import com.example.apublic.CommonCallBack
import com.example.pearvideoclient.Constants
import com.example.pearvideoclient.R
import com.example.pearvideoclient.channel.adapter.ContListAdapter
import com.example.pearvideoclient.entity.ContEntity
import com.scwang.smartrefresh.layout.api.RefreshLayout
import kotlinx.android.synthetic.main.fragment_category_conts.*

import java.util.ArrayList

/**
 * @Description: java类作用描述
 * @Author: jiazhu
 * @Date: 2019-08-29 15:51
 * @ClassName: CategoryFragment
 */
class CategoryFragment : Fragment() {
    private val mContListAdapter = ContListAdapter(ArrayList())

    private var refreshCallBack: CommonCallBack<Int, Unit>? = null

    fun setRefreshCallBack(refreshCallBack: CommonCallBack<Int, Unit>) {
        this.refreshCallBack = refreshCallBack
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_category_conts, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initData()
    }

    private fun initData() {
        val activity = activity as AppCompatActivity?
        rv_category_cont_list.layoutManager = LinearLayoutManager(activity, RecyclerView.VERTICAL, false)
        rv_category_cont_list.adapter = mContListAdapter

        refresh_layout.setOnRefreshListener {
            refreshCallBack?.todo(Constants.LOAD_REFRESH)
        }

        refresh_layout.setOnLoadMoreListener {
            refreshCallBack?.todo(Constants.LOAD_MORE)
        }

    }

    fun replaceData(data: List<ContEntity>) {
        mContListAdapter.replaceData(data)
    }

    fun addData(data: List<ContEntity>) {
        mContListAdapter.addData(data)
    }

    fun loadMoreFinish(isSuccess: Boolean) {
        refresh_layout.finishLoadMore(isSuccess)
    }

    fun refreshFinish(isSuccess: Boolean) {
        refresh_layout.finishRefresh(isSuccess)
    }

    fun hasData(): Boolean {
        return mContListAdapter.data.size > 0
    }

    companion object {
        @JvmStatic
        fun newInstance(name: String, categoryId: String): CategoryFragment {
            val bundle = Bundle()
            bundle.putString("name", name)
            bundle.putString("categoryId", categoryId)
            val fragment = CategoryFragment()
            fragment.arguments = bundle
            return fragment
        }
    }
}
